test/box-luatest/gh_8204_fast_offset_test.lua

```
@@ -68,14 +102,844 @@ g.test_option = function()
+            if tx_arg == nil then
+                tx:begin()
+            end
+
+            check(tx, it, key, expected_count, file, line)
+            for _, new_key in pairs(unexisting_keys) do
+                s:insert(new_key)
+                check(tx, it, key, expected_count, file, line)
+            end
+            for _, old_key in pairs(existing_keys) do
+                s:delete(old_key)
+                check(tx, it, key, expected_count, file, line)
+            end
+            for _, old_key in pairs(unexisting_keys) do
+                s:delete(old_key)
+                check(tx, it, key, expected_count, file, line)
+            end
+            for _, new_key in pairs(existing_keys) do
+                s:insert(new_key)
+                check(tx, it, key, expected_count, file, line)
+            end
+
+            -- Autocommit if no transaction passed.
+            if tx_arg == nil then
+                t.assert_equals(tx:commit(), success)
+            end
+
+            t.assert_equals(s:len(), old_len)
+            t.assert_equals(s:select(), existing_keys)
+        end
+
+        -- No conflict with itself (count all in empty & insert).
+        for _, it in pairs(all_iterators) do
+            count_do(tx, tx, it, {}, 0, 'insert', {37, 37}, success)
+        end
+
+        -- No conflict with itself (count empty by key & insert matching).
+        count_do(tx, tx, 'lt', {0, 1}, 0, 'insert', {0, 0}, success)
+        count_do(tx, tx, 'le', {0, 0}, 0, 'insert', {0, 0}, success)
+        count_do(tx, tx, 'req', {0, 0}, 0, 'insert', {0, 0}, success)
+        count_do(tx, tx, 'eq', {0, 0}, 0, 'insert', {0, 0}, success)
+        count_do(tx, tx, 'ge', {0, 0}, 0, 'insert', {0, 0}, success)
+        count_do(tx, tx, 'gt', {0, 0}, 0, 'insert', {0, 1}, success)
+
+        -- No conflict with itself (count empty by key & insert not matching).
+        count_do(tx, tx, 'lt', {0, 0}, 0, 'insert', {0, 0}, success)
```

The insert silently fails, it must be a replace.